### PR TITLE
Example cucumber test

### PR DESCRIPTION
**Pull request by stsefanenko**

### CODE REVIEW

File: examples/cucumber/pom.xml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>cucumber</artifactId>
+
+    <dependencyManagement>
+        <dependencies>
+            <dependency>
+                <groupId>org.springframework.boot</groupId>
+                <artifactId>spring-boot-dependencies</artifactId>
```
Comment (bsideup) [inline examples/cucumber/pom.xml]: this example does not use Spring Boot at all

File: examples/cucumber/src/test/java/org/testcontainsers/examples/Stepdefs.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import org.openqa.selenium.remote.RemoteWebDriver;
+import org.testcontainers.containers.BrowserWebDriverContainer;
+
+import java.io.File;
+import java.util.List;
+
+import static junit.framework.TestCase.assertEquals;
+import static org.testcontainers.containers.BrowserWebDriverContainer.VncRecordingMode.RECORD_ALL;
+
+public class Stepdefs {
```
Comment (bsideup) [inline examples/cucumber/src/test/java/org/testcontainsers/examples/Stepdefs.java]: Please also integrate with `TestLifecycleAware` interface and make it record video after a test failure (by calling `afterTest` with non-empty `throwable` argument

File: examples/cucumber/pom.xml (hunk 1 above)
Comment (stsefanenko) [reply examples/cucumber/pom.xml]: removed

File: examples/cucumber/src/test/java/org/testcontainsers/examples/Stepdefs.java (hunk 2 above)
Comment (stsefanenko) [reply examples/cucumber/src/test/java/org/testcontainsers/examples/Stepdefs.java]: added

File: examples/cucumber/src/test/java/org/testcontainsers/examples/Stepdefs.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+                @Override
+                public String getTestId() {
+                    return scenario.getId();
+                }
+
+                @Override
+                public String getFilesystemFriendlyName() {
+                    return scenario.getName();
+                }
+            }, Optional.of(RuntimeException.class));
```
Comment (bsideup) [inline examples/cucumber/src/test/java/org/testcontainsers/examples/Stepdefs.java]: this should be `Optional<Throwable>` not `Optional<Class>`

Comment (stsefanenko) [reply examples/cucumber/src/test/java/org/testcontainsers/examples/Stepdefs.java]: fixed parameter

File: examples/cucumber/src/test/java/org/testcontainsers/examples/Stepdefs.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+                    return scenario.getId();
+                }
+
+                @Override
+                public String getFilesystemFriendlyName() {
+                    return scenario.getName();
+                }
+            }, Optional.of(new RuntimeException()));
+        }
+        container.stop();
```
Comment (bsideup) [inline examples/cucumber/src/test/java/org/testcontainsers/examples/Stepdefs.java]: `afterTest` will call `stop`, no need to do it here. It's better to always call `afterTest`, with the following parameter:
```java
Optional.of(scenario).filter(Scenario::isFailed).map(__ -> new RuntimeException())
```

File: examples/pom.xml [hunk 5]
```diff
@@ -70,6 +71,8 @@
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <testcontainers.group>com.github.testcontainers.testcontainers-java</testcontainers.group>
         <testcontainers.version>-SNAPSHOT</testcontainers.version>
+        <cucumber.vesrion>2.3.1</cucumber.vesrion>
```
Comment (bsideup) [inline examples/pom.xml]: no need to put it here, it is not shared between all examples

Comment (stsefanenko) [reply examples/pom.xml]: updated

File: examples/cucumber/src/test/java/org/testcontainsers/examples/Stepdefs.java (hunk 4 above)
Comment (stsefanenko) [reply examples/cucumber/src/test/java/org/testcontainsers/examples/Stepdefs.java]: fixed

**Comment by rnorth** [comment]
We'll need to update this once #1177 is merged, unfortunately.

**Comment by stsefanenko** [comment]
Updated to gradle

**Comment by bsideup** [comment]
Thanks @stsefanenko! Sorry for a lack of attention on this one. I looks good, I will have a final look and merge after CI reports green status 👍

File: examples/cucumber/src/test/java/org/testcontainsers/examples/Stepdefs.java [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+import java.util.Optional;
+
+import static junit.framework.TestCase.assertEquals;
+import static org.testcontainers.containers.BrowserWebDriverContainer.VncRecordingMode.RECORD_ALL;
+
+public class Stepdefs {
+
+    private BrowserWebDriverContainer container = new BrowserWebDriverContainer()
+            .withCapabilities(new ChromeOptions())
+            .withRecordingMode(RECORD_ALL, new File("target"));
```
Comment (bsideup) [inline examples/cucumber/src/test/java/org/testcontainsers/examples/Stepdefs.java]:
```suggestion
            .withRecordingMode(RECORD_ALL, new File("build"));
```

**Comment by bsideup** [comment]
@stsefanenko merged! 🎉 
Thanks a lot, good job! 👍

**Comment by rnorth** [comment]
Released in https://github.com/testcontainers/testcontainers-java/releases/tag/1.10.7!

**Pull request merged**